tests/e2e/scale/noobaa/test_list_objects.py

```diff
@@ -98,3 +103,66 @@ def test_list_large_number_of_objects(
             f"Files count {files_count} and objects_count {objects_count} does not match. "
             "Check does all objects are listed correctly!!!"
         )
+
+    @pytest.fixture()
+    def setup_dir(self, request, awscli_pod_session, test_directory_setup):
+        """
+        This is to setup 1000s of directories where in each directory has atleast 10 objects
+        and one directory contains almost 3 Mil objects
+        """
+        origin_dir = test_directory_setup.origin_dir
+
+        # create 1000 directories under origin_dir
+        amount = 1000
+        command = (
+            f"for i in $(seq 0 {amount-1});"
+            f"do mkdir {origin_dir}/dir_$i && chmod -R 777 {origin_dir}/dir_$i; done"
+        )
+        awscli_pod_session.exec_sh_cmd_on_pod(command=command, sh="sh", timeout=9000)
+
+        # create 10s and millions of objects under the directories created above
+        command = (
+            f"for i in $(seq 0 {amount-1});"
+            f"do obj=10;"
+            f"if [ $i == 500 ];"
+            f"then obj=3000000; fi;"
+            f"for j in $(seq 0 $obj);"
+            f"do dd if=/dev/urandom of={origin_dir}/dir_$i/obj_key$j bs=512 count=1 status=none; done; done"
+        )
+        awscli_pod_session.exec_sh_cmd_on_pod(command=command, sh="sh", timeout=9000)
+        log.info(
+            f"All the directories & objects are created in parent directory {origin_dir} !"
+        )
+
+        def teardown():
+            awscli_pod_session.exec_cmd_on_pod(command=f"rm -rf {origin_dir}/*")
+
+        request.addfinalizer(teardown)
+        return origin_dir, "dir_500"
+
+    @bugzilla("2110504")
+    @pytest.mark.polarion_id("OCS-4650")
+    def test_list_with_prefix_delimiter(
+        self, bucket_factory, setup_dir, awscli_pod_session, mcg_obj
+    ):
+        """
+        This will test list directory with large number of objects using prefix & delimeter
+        """
+        origin_dir, large_dir = setup_dir
+        bucket_name = bucket_factory()[0].name
+
+        # sync all the directories and objects to the bucket
+        sync_object_directory(
+            awscli_pod_session, f"{origin_dir}", f"s3://{bucket_name}", s3_obj=mcg_obj
+        )
+        log.info("Synced all the object directories!")
+
+        # list objects with prefix and delimeter
+        list_objects_from_bucket(
```

Comment:
How many objs are expected to list here? You might have to think about page size if it is over 1000 objs.
https://docs.aws.amazon.com/cli/latest/reference/s3/ls.html

Reply:
Update: Max allowed page size is 1000, so cant this cant be listed more than 1000 obj. Might have to use boto

Reply:
@parikshithb I guess page size defines maximum number of output within single response. But a single `s3 ls` request will have multiple responses untill all the objects are listed. so it doesnt really affect as per my testing. 
For example, here I set the page size 10 so that each response have only 10 ouput but it still lists out all the objects with multiple responses:

> $ s3 ls s3://s3-bucket-368dd4efaccb47039e265864200f60/dir_50/ --page-size 10
> /usr/lib/python3.6/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
>   InsecureRequestWarning)
> 2022-09-27 11:37:43        512 obj_key0
> 2022-09-27 11:37:43        512 obj_key1
> 2022-09-27 11:37:43        512 obj_key10
> 2022-09-27 11:37:43        512 obj_key100
> 2022-09-27 11:37:43        512 obj_key1000
> 2022-09-27 11:37:43        512 obj_key1001
> 2022-09-27 11:37:43        512 obj_key1002
> 2022-09-27 11:37:43        512 obj_key1003
> 2022-09-27 11:37:43        512 obj_key1004
> 2022-09-27 11:37:43        512 obj_key1005
> /usr/lib/python3.6/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
>   InsecureRequestWarning)
> 2022-09-27 11:37:44        512 obj_key1006
> 2022-09-27 11:37:44        512 obj_key1007
> 2022-09-27 11:37:44        512 obj_key1008
> 2022-09-27 11:37:44        512 obj_key1009
> 2022-09-27 11:37:44        512 obj_key101
> 2022-09-27 11:37:44        512 obj_key1010
> 2022-09-27 11:37:44        512 obj_key1011
> 2022-09-27 11:37:44        512 obj_key1012
> 2022-09-27 11:37:44        512 obj_key1013
> 2022-09-27 11:37:44        512 obj_key1014
> /usr/lib/python3.6/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
>   InsecureRequestWarning)
> 2022-09-27 11:37:44        512 obj_key1015
> 2022-09-27 11:37:44        512 obj_key1016
> 2022-09-27 11:37:44        512 obj_key1017
> 2022-09-27 11:37:44        512 obj_key1018
> 2022-09-27 11:37:44        512 obj_key1019
> 2022-09-27 11:37:44        512 obj_key102
> 2022-09-27 11:37:44        512 obj_key1020
> 2022-09-27 11:37:44        512 obj_key1021
> 2022-09-27 11:37:44        512 obj_key1022
> 2022-09-27 11:37:44        512 obj_key1023
> /usr/lib/python3.6/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
>   InsecureRequestWarning)
>